src/Toolkit/Toolkit/UI/Controls/SearchView/SearchViewModel.cs

```diff
@@ -0,0 +1,544 @@
+// /*******************************************************************************
+//  * Copyright 2012-2018 Esri
+//  *
+//  *  Licensed under the Apache License, Version 2.0 (the "License");
+//  *  you may not use this file except in compliance with the License.
+//  *  You may obtain a copy of the License at
+//  *
+//  *  http://www.apache.org/licenses/LICENSE-2.0
+//  *
+//  *   Unless required by applicable law or agreed to in writing, software
+//  *   distributed under the License is distributed on an "AS IS" BASIS,
+//  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  *   See the License for the specific language governing permissions and
+//  *   limitations under the License.
+//  ******************************************************************************/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Tasks.Geocoding;
+
+namespace Esri.ArcGISRuntime.Toolkit.UI.Controls
+{
+    /// <summary>
+    /// Backing controller for a search experience, intended for use with SearchView.
+    /// SearchView supports searching, with search-as-you-type for multiple search providers via <see cref="ISearchSource"/>.
+    /// </summary>
+    public class SearchViewModel : INotifyPropertyChanged
+    {
+        private const int QueryTimeoutMilliseconds = 2000;
+        private ISearchSource? _activeSource;
+        private SearchResult? _selectedResult;
+        private string? _currentQuery;
+        private string? _defaultPlaceholder = "Find a place or address";
```

Comment:
Should support for localization with resx be included in new controls as MVP?

Reply:
Yes we already have a resx file you can use https://github.com/Esri/arcgis-toolkit-dotnet/blob/main/src/Toolkit/Toolkit/LocalizedStrings/Resources.resx

Reply:
I've filed an issue to track this work: https://github.com/Esri/arcgis-toolkit-dotnet/issues/386